Add clarifications for string key reliable collection sort order (#115221)
Azure/azure-content: articles/service-fabric/service-fabric-reliable-services-reliable-collections-guidelines.md

  Only one user thread operation is supported within a transaction. Otherwise, it will cause memory leak and lock issues.
* Consider dispose transaction as soon as possible after commit completes (especially if using ConcurrentQueue).
* Do not perform any blocking code inside a transaction.

> * When [string](/dotnet/api/system.string) is used as the key for a reliable dictionary, the sorting order will be using [default string comparer CurrentCulture](/dotnet/api/system.string.compare#system-string-compare(system-string-system-string)). Please note that the CurrentCulture sorting order is different from [Ordinal string comparer](/dotnet/api/system.stringcomparer.ordinal). 

Reviewer: ```suggestion
* When [string](/dotnet/api/system.string) is used as the key for a reliable dictionary, the sorting order uses [default string comparer CurrentCulture](/dotnet/api/system.string.compare#system-string-compare(system-string-system-string)). Note that the CurrentCulture sorting order is different from [Ordinal string comparer](/dotnet/api/system.stringcomparer.ordinal). 
```
